rpcs3/Emu/Cell/lv2/sys_net.cpp

```diff
@@ -1142,9 +1221,23 @@ error_code sys_net_bnet_recvfrom(ppu_thread& ppu, s32 s, vm::ptr<void> buf, u32
 		paddr->sin_port   = ntohs(reinterpret_cast<struct sockaddr_in*>(&native_addr)->sin_port);
 		paddr->sin_addr   = ntohl(reinterpret_cast<struct sockaddr_in*>(&native_addr)->sin_addr.s_addr);
 		paddr->sin_zero   = 0;
+
+		if (type == SYS_NET_SOCK_DGRAM_P2P)
+		{
+			vm::ptr<sys_net_sockaddr_in_p2p> paddr_p2p = vm::cast(addr.addr());
+			paddr_p2p->sin_vport                       = paddr_p2p->sin_port - 3658;
+			paddr_p2p->sin_port                        = 3658;
+
+			u16 daport  = reinterpret_cast<sys_net_sockaddr_in*>(addr.get_ptr())->sin_port;
+			u16 davport = reinterpret_cast<sys_net_sockaddr_in_p2p*>(addr.get_ptr())->sin_vport;
```

Comment:
const const

Reply:
you don't need reinterpret_cast if you had use vm::_ptr instead of vm::ptr.